src/features/userSlice.ts

```diff
@@ -0,0 +1,32 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { User } from '../types/User';
+import { getUsers } from '../api/users';
+
+export type UserState = {
+  users: User[];
+};
```

Comment:
Consider adding `loading` and `err` states to the `UserState` type to handle the asynchronous fetching process more effectively.